### DISCUSSION TITLE
Can I add new fields to websocket?

### DISCUSSION THREAD

**Discussion by yilia-jcq** [eclipse-emfcloud/emfcloud#148] [Coffee Editor]
Can I add new fields to websocket? If so, what does the json format look like? Where should I add it?

**Comment by eneufeld** [comment]
Hi @yilia-jcq ,

I can't follow you, can you elaborate what you want to do?

**Reply by yilia-jcq** [reply]
Dear @eneufeld ,
For example, the websocket send field in the following figure has "id, kind, content", can I expand a new field on this basis?

![image](https://user-images.githubusercontent.com/101612117/158718296-8b8392b5-8b4b-463c-b0a8-f606a94e9cb6.png)

**Reply by eneufeld** [reply]
Could you elaborate which communication this is, so talking to which service? Also why do you want to add more fields to it?

**Reply by yilia-jcq** [reply]
Dear @eneufeld ，
Now in coffee-editor, the websocket send request format is as follows：
![image](https://user-images.githubusercontent.com/101612117/158811497-9729c2eb-04bd-4bc3-9bc2-222e27c6801a.png)

I want to add a parameter to the send method in the "web/browser-app/lib/bundle.js" file, such as "customerId", can I write it like 
`{kind: 'data',id: this.id,content: content,customerId：2}`

**Reply by CamilleLetavernier** [reply]
The Coffee Editor is an example application that aggregates several related or unrelated components.

When the Model Server is involved, we use REST API Calls, rather than Json RPC. I think GLSP Diagrams rely on Json RPC messages.

In both cases, messages are not defined directly by Json RPC; we define our own Action or Message format, somewhere in the "content" attribute. So, as far as I know, we never add additional fields to the Json RPC message directly; but rather somewhere in the content attribute (But we really abstract Json RPC even in GLSP; so we almost never have to deal with these low-level messages directly).

Without knowing what you're trying to do or which specific component(s) you're working with (GLSP Diagram, Tree Editor, Model Server, Properties, other?), it's hard to give an accurate answer.

**Reply by yilia-jcq** [reply]
Dear @CamilleLetavernier ,

First of all, I was very happy to receive your reply, and it made me very confused, and now my project is doing a model server.

Replying above, you said that the model server you are using the REST API call, so if it is a call method, can I see http requests in the network, but now I can only see websocket requests in the network.

So can I understand your http request this way, is the ui page to the websocket to node .js, and then node forwarded to http request to the backend?
